openbsd/OpenBSDMachine.c

```diff
@@ -273,7 +262,7 @@ static void OpenBSDMachine_scanCPUTime(OpenBSDMachine* this) {
 }
 
 void Machine_scan(Machine* super) {
-   OpenBSDMachine* this = (OpenBSDMachine*) super;
+   OpenBSDMachine* this = super;
```

Comment:
I didn't know this cast was here initially. Let me know if you want me to add it back. Though removing it didn't mess up the build. I might redo this pr from scratch if i mess up too much.

Reply:
The rule usually is casting from generic to specific (platform) type; no casting from platform type to generic (use `->super` instead).

Also: Feel free to rebase/squash the commits to your hearts content, until they look clean.

Reply:
After I'm done making all my changes locally to how we want the final result to look, i will create a new pr, close this one, and break it up into a couple commits, ones that have code changes related to one another in how it is broken up, without the excess in extra commits. Is that ok?

In the meantime, feel free to let me know how you want the commits broken up here.

Reply:
Please instead of opening a new PR, just split the changes into commits as appropriate and push them (`--force-with-lease`) on the current branch.

Regarding splitting the commits:
- One commit (or more, if there are vastly distinct build issues) for compile issues
- One commit fixing the memory detection (see my other comment regarding that)

Apart from that, providing descriptive commit messages and following the information outlined in the style guide is much appreciated.